ouroboros-network/src/Ouroboros/Network/Protocol/Handshake/Version.hs

```diff
@@ -60,6 +61,18 @@ simpleSingletonVersions vNum vData extra r =
 data Sigma f where
   Sigma :: t -> f t -> Sigma f
 
+
+-- |
+-- A @'Maybe'@ like type which better explains its purpose.
+--
+data Accept
+  = Accept
+  | Refuse !Text
+  deriving (Eq, Show)
+
+class Acceptable v where
+  acceptV :: v -> v -> Accept
```

Comment:
let's use 
```
acceptVersion :: versionParameters -> versionParameters -> Accept
```
Also this class, as part of public api, needs to be well documented: what is the intention how it is used, what are the two parameters, etc.

Reply:
Just an idea:
If instead of a type class we'd embed a `v -> v -> Accept` function, this would give us more flexibility how versions are accepted.  With a type class every time we use `v` we end-up with the same `Acceptable` instance.

Reply:
And it would also be easier for testing - no need to write instances for custom types.  

The expense is that `simpleSingletonVersion` would get additional argument.

Reply:
There already is such a function for the responder side, see https://github.com/input-output-hk/ouroboros-network/blob/0841d21fa05cd02b2f606b62b0241edbca6138b7/ouroboros-network/src/Ouroboros/Network/Protocol/Handshake/Type.hs#L246 .
This pr only changes `Eq` instance which is used by the handshake test cases client to verify that the server managed to pick a correct version.
I'm not sure what to do with `simpleSingletonVersion`  yet, adding a function for two comparing two fields it currently ignores seems premature.

Reply:
Yes, we can easily change it later, when / if needed.